http.go

```diff
@@ -14,8 +14,9 @@ import (
 
 // Response is a struct to generate a response from POST/PUT requests
 type Response struct {
-	Code int
-	Body interface{}
+	Code    int
+	Body    interface{}
+	Headers map[string]string
```

Comment:
The header order is important under certain special cases. In particular, when issuing the same header multiple times (which is valid, but not supported on this implementation).
https://stackoverflow.com/questions/750330/does-the-order-of-headers-in-an-http-response-ever-matter

We need to, at least, document that we don't support multiple headers and that the order is unspecified.

Reply:
Added comments